vault/resource_ldap_secret_backend_static_role.go

```diff
@@ -64,9 +69,21 @@ var ldapSecretBackendStaticRoleFields = []string{
 	consts.FieldUsername,
 	consts.FieldDN,
 	consts.FieldRotationPeriod,
+	consts.FieldSkipImportRotation,
+}
+
+func updateLDAPStaticRoleResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Vault rejects an update request containing skip_import_rotation since the field has no effect.
+	// maybe Vault should silently ignore it, but for now just remove the offending field
+	err := d.Set(consts.FieldSkipImportRotation, nil)
```

Comment:
Thinking out loud here: I just realized this was being set to nil. I am not sure about this but it looks like the ldap API does not return `skip_import_rotation` from the GET enpoint. In which case I don't think we need to worry about drift in TF's state vs Vault's state. So this is probably ok as-is?

Reply:
my worry was that if a terraform config includes the field, we need to strip it out or vault will reject it if an update action is triggered (say, if the element is modified?)

In retrospect Vault itself probably should _at most_ warn. <kirbyshrug>

Reply:
Yeah, Vault usually ignores irrelevant fields without even a warning, which is a bit strange. I agree that we should probably warn instead of error.

Setting to nil is probably ok as-is with the caveat that there may be situations I haven't considered. 

Another option might be to have separate Create and Update functions and only send `skip_import_rotation` in the Create request. Then we could leave the TF state unaltered. I don't think it is very common to use the TF state as the place where we store the request data. TF state can (and often does) include entries that are not included in the API requests.